translation/translation.js

```diff
@@ -0,0 +1,271 @@
+            fetch('../../../../../../concerteaux3d/translation.json').then(function(response) {
+                if (response.ok)
+                    return response.json();
+                throw new Error('Network response was not ok.');
+                }).then(function(translation) {
+
+                    // put the json content in to the dictionary
+                    translationIT_FR2 = translation;
+                    //console.log(translationIT_FR2);
+                    // iterate over dictionary key
+                    Object.keys(translationIT_FR2).forEach((italian) => {
+                        //check if a translation is provided for each string
+                        if (translationIT_FR2[italian] != ""){
+                            // translate layers name in the edition form
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container div h3").filter(function(){
+                                return $(this).text() === italian
+                            }).html(translationIT_FR2[italian]);
+                            // translate aliases or fields name in the edition form
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container form#jforms_view_edition.form-horizontal div.control-group label").filter(function(){
+                                return $(this).text() === italian
+                            }).html(translationIT_FR2[italian]);
+                        }
+                        else {
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container div h3").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container form#jforms_view_edition.form-horizontal div.control-group label").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                        }
+                    });
+            });
+
+        }
+    }
+});
+
+//This function is executed when the lizmappopupdisplayed event is triggered
+lizMap.events.on({
+    lizmappopupdisplayed: function(e) {
+        // get browser language
+        var userLang = navigator.language || navigator.userLanguage;
+        if (userLang === 'fr' || userLang === 'fr-FR'){
+
+            var translationIT_FR3 = {};
+
+            // get the json file with transalted strings
+            // the path to the json file must be provided as relative path, using the web url the browser can return a security error
+            fetch('../../../../../../concerteaux3d/translation.json').then(function(response) {
```

Comment:
You could cache translation.js as you have download it with "uicreated" event above.

Reply:
I have changed the code, now the fetch of the translation.json file is done only once in the "uicreated" event.